modules/plan.nix

```diff
@@ -15,8 +15,16 @@
 with lib;
 with types;
 
-
 let
+  # dealing with str is a bit annoying espectially with `nullOr str` as that apparently defaults to ""
+  # instead of null :shrug:.  This then messes with our option inheritance logic.
+  # Hence we have a uniqueStr type that ensures multiple identially defined options are collapsed
+  # without raising an error. And a way to fetch default options that will retain `null` if the
+  # option is not defined or "".
+  getDefaultOrNull = def: key: if def ? ${key} && def.${key} != "" then def.${key} else null;
+  mergeUniqueOption = locs: defs: mergeOneOption locs (lists.unique defs);
```

Comment:
What about the case when merging `"a string"` and `null`? Should we filter out null from defs if its length is >= 2 ?

Reply:
I was wondering about this as well, but I'd rather see this happening first.